ci/.gitlab-ci.yml

```diff
@@ -33,12 +30,37 @@ stages:
     - TAG_REPO=`find $SPACK_DLAF_REPO -type f -exec sha256sum {} \; | sha256sum - | head -c 16`
     - TAG_ENVIRONMENT=`sha256sum $SPACK_ENVIRONMENT | head -c 16`
     - TAG=${TAG_IMAGE}-${TAG_APTGET}-${TAG_COMPILER}-MKL${USE_MKL}-${TAG_DOCKERFILE}-${TAG_SPACK}-${TAG_REPO}-${TAG_ENVIRONMENT}
-    - (docker pull $BUILD_IMAGE:$TAG && docker tag $BUILD_IMAGE:$TAG $BUILD_IMAGE:latest) || docker build -t $BUILD_IMAGE:$TAG -t $BUILD_IMAGE:latest --cache-from $BUILD_IMAGE:latest --build-arg BASE_IMAGE --build-arg BUILDKIT_INLINE_CACHE=1 --build-arg SPACK_SHA --build-arg EXTRA_APTGET --build-arg COMPILER --build-arg CXXSTD --build-arg SPACK_ENVIRONMENT --build-arg SPACK_DLAF_REPO --build-arg USE_MKL -f $BUILD_DOCKER_FILE --network=host .
-    - docker push $BUILD_IMAGE:$TAG
-    - docker push $BUILD_IMAGE:latest
-    - docker build -t $DEPLOY_IMAGE --build-arg BUILD_IMAGE=$BUILD_IMAGE:$TAG --build-arg DEPLOY_BASE_IMAGE --build-arg EXTRA_APTGET_DEPLOY --build-arg USE_MKL --build-arg USE_ROCBLAS -f $DEPLOY_DOCKER_FILE --network=host .
-    - docker push $DEPLOY_IMAGE
-    - docker run -v $PWD/ci/ctest_to_gitlab.sh:/ctest_to_gitlab.sh $DEPLOY_IMAGE /ctest_to_gitlab.sh "$DEPLOY_IMAGE" "$USE_CODECOV" "$THREADS_PER_NODE" "$SLURM_CONSTRAINT" > pipeline.yml
+    - mount -t tmpfs none /var/tmp
+    - mount -t tmpfs none /var/lib/containers
+    - '(podman pull $BUILD_IMAGE:$TAG && podman tag $BUILD_IMAGE:$TAG $BUILD_IMAGE:latest)
+       ||  podman build
+           -t $BUILD_IMAGE:$TAG -t $BUILD_IMAGE:latest
+           --format docker --layers --isolation chroot
+           --build-arg BASE_IMAGE
+           --build-arg BUILDKIT_INLINE_CACHE=1
+           --build-arg SPACK_SHA
+           --build-arg EXTRA_APTGET
+           --build-arg COMPILER
+           --build-arg CXXSTD
+           --build-arg SPACK_ENVIRONMENT
+           --build-arg SPACK_DLAF_REPO --build-arg USE_MKL
+           --build-arg NUM_PROCS=$(echo "$(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us) $(cat /sys/fs/cgroup/cpu/cpu.cfs_period_us) $(nproc)" | awk ''{ print ($1/$2 > 0) ? int($1/$2+0.5) : $3; }'')
```

Comment:
As mentioned by @finkandreas, we should pass `NUM_PROCS` to `spack install` (the nodes have 256 cores (/hyperthreads?) but only 64 are allocated per pod. However, `nproc` reports 256 and the `/sys/fs/cgroup` stuff reports 64 (@finkandreas please correct me if I misunderstood).

I can push this change.

Reply:
You are absolutely right. NUM_PROCS works as is on both bare metal and inside a cgroup environment. `nproc` would give the physical amount of cores, independent what the k8s restrictions are. On bare metal `/sys/fs/cgroup/cpu/cpu.cfs_quota_us` is typically `-1`, i.e. we fallback to `nproc`.
This variable should of course be used inside the Dockerfile / spack environment file under the key `config:build_jobs`, or if you use the new Makefile approch then start installation of spack packages with `make -j $NUM_PROCS`.

Reply:
I have a couple of question as I'm afraid we can go easily out-of-memory...
I suppose that they are AMD nodes with two 64-cores sockets (x2 SMT).
How much memory do these nodes have?
64 cores are allocated per pod. Does this mean 32 real cores (x2 SMT) or 64 real cores? Or stated in a different way: are 2 or 4 pods per node allowed?

Reply:
My concern the cores being too much, but memory that can be too few:
512GB means 2GB/thread and it is not much as we are also building in memory.
For images (which now are built in memory) I count ~5GB and when building DLAF with debug symbols each process needs more than 2GB. (once I was testing the dockerfile and make -j 4 was failing with 8GB of ram...).
So we can easily consume a lot of memory if many of our jobs are put on the same node with few cores (therefore less available memory) each.